keps/sig-cli/NNNN-kustomize-components.md

```diff
@@ -0,0 +1,676 @@
+  - [User Story](#user-story)
+  - [Notes/Constraints/Caveats (optional)](#notesconstraintscaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+(**FIXME:** Explain what is the current situation and then roughly explain what this feature does)
+
+> This section is incredibly important for producing high quality user-focused
+> documentation such as release notes or a development roadmap.  It should be
+> possible to collect this information before implementation begins in order to
+> avoid requiring implementors to split their attention between writing release
+> notes and implementing the feature itself.  KEP editors, SIG Docs, and SIG PM
+> should help to ensure that the tone and content of the `Summary` section is
+> useful for a wide audience.
+>
+> A good summary is probably at least a paragraph in length.
+
+## Motivation
+
+By design, Kustomize and its variant system follow an inheritance model rather
+than a composition one. This poses certain limitations in complex deployment
+scenarios that deal with multivariate applications, i.e., ones that:
+
+- ship with numerous opt-in features and corresponding configuration options
+- target dissimilar audiences with different needs and scopes (developers,
+  admins, customers, testers etc.)
+- run on multiple cloud platforms (on-prem, Minikube, GKE, EKS, AKS, etc.)
+- tailor semantically distinct parts of a
+  common, parent configuration simultaneously
+
+Currently, declarative management of such applications with stock Kubernetes
+CLI tools is quite hard and can quickly get out of control.
```

Comment:
> stock Kubernetes CLI tools

I think we can just say Kustomize here

> and can quickly get out of control.

I think you need to explain why here. Basically, move the "The reason for is is that some applications..." part from the paragraph below.

Reply:
> I think we can just say Kustomize here

Yes, I was mainly referring to `kustomize` and `kubectl` without other external tools or plugins.